### ISSUE TITLE
Custom services broken due to missing CORS permission in manifest.json

### ISSUE THREAD

**Issue by nitroxis** [fwenzel/copy-shorturl#72]
Hi,
The extension is currently not working for custom service URLs because of missing permissions. This is what I see in the browser console:
```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://example.org. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).
TypeError: NetworkError when attempting to fetch resource.  bundle.js:164:15
```

It seems you've already added the "built-in" services to the manifest.json, but a custom URL would probably have to be added as well. Not sure if this can be done at runtime though. One workaround would be to simply [allow all URLs](https://stackoverflow.com/questions/36629635/firefox-webextensions-and-cross-domain-privileges) like this:
```
manifest.json
{
  ...
  "permissions": [
    "<all_urls>" 
  ]
}
```

**Comment by fwenzel** [comment]
Oh thank you, this is a very good catch. While I don't like having to open up permissions so widely, I figure an extension like this can justifiably have this permission. Also, it's open source and therefore people can just read the source code if they are worried.

How urgent is this for you? I was going to look into adding bit.ly support as well (if I can get an API key to test, without having to pay them). Maybe I can roll that into the same change.

**Comment by nitroxis** [comment]
It's not that urgent. But thanks for fixing it!

**Comment by fwenzel** [comment]
I fixed the bug just now, added bitly support, and released version 4.2 on addons.mozilla.org -- when it's approved, your browser should update you to the latest version automatically!

Thanks again for filing this issue!

**Comment by fwenzel** [comment]
aaand it's approved. Enjoy!
